util/vitess/vitess_hash.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vitess
+
+import (
+	"crypto/cipher"
+	"crypto/des"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/pingcap/errors"
+)
+
+var nullKeyBlock cipher.Block
+
+func init() {
+	block, err := des.NewCipher(make([]byte, 8))
+	if err != nil {
+		panic(errors.Trace(err))
+	}
+	nullKeyBlock = block
+}
+
+// Hash implements vitess' method of calculating a hash used for determining a shard key range.
+// Uses a DES encryption with 64 bit key, 64 bit block, null-key
+func Hash(shardKey []byte) (uint64, error) {
+	if len(shardKey) > 8 {
```

Comment:
We need test cases to cover this branch.

Reply:
Deleted this branch instead.